tensorboard/webapp/feature_flag/views/feature_flag_page_test.ts

```diff
@@ -0,0 +1,232 @@
+          flags: {
+            inColab: false,
+          },
+        })
+      );
+    });
+
+    it('creates removes override when status is set to default', () => {
+      store.overrideSelector(getDefaultFeatureFlags, {
+        inColab: false,
+      } as FeatureFlags);
+      store.overrideSelector(getOverriddenFeatureFlags, {});
+      createComponent();
+      const matSelect = fixture.debugElement.query(By.css('mat-select'));
+      matSelect.triggerEventHandler('selectionChange', {value: 'default'});
+
+      expect(dispatchSpy).toHaveBeenCalledWith(
+        featureFlagOverridesReset({
+          flags: ['inColab'],
+        })
+      );
+    });
+  });
+
+  describe('onAllFlagsReset', () => {
+    it('resets all feature flags', () => {
+      store.overrideSelector(getDefaultFeatureFlags, {} as FeatureFlags);
+      store.overrideSelector(getOverriddenFeatureFlags, {});
+      createComponent();
+      const button = fixture.debugElement.query(By.css('button'));
+      button.triggerEventHandler('click', {});
+      expect(dispatchSpy).toHaveBeenCalledWith(allFeatureFlagOverridesReset());
+    });
+  });
+
+  describe('getFlagStatus', () => {
+    it('returns default when flag is not overridden', () => {
+      const status = TEST_ONLY.getFlagStatus('inColab', {});
+      expect(status).toEqual(FeatureFlagOverrideStatus.DEFAULT);
+    });
+
+    it('returns enabled when flag is overridden to true', () => {
+      const status = TEST_ONLY.getFlagStatus('inColab', {
+        inColab: true,
+      });
+      expect(status).toEqual(FeatureFlagOverrideStatus.ENABLED);
+    });
+
+    it('returns disabled when flag is overridden to false', () => {
+      const status = TEST_ONLY.getFlagStatus('inColab', {
+        inColab: false,
+      });
+      expect(status).toEqual(FeatureFlagOverrideStatus.DISABLED);
+    });
+  });
+
+  // Tests for feature_flag_page_component
+  it('creates rows for each feature flag', () => {
+    store.overrideSelector(getDefaultFeatureFlags, {
+      inColab: false,
+      enabledExperimentalPlugins: [] as string[],
+    } as FeatureFlags);
+    store.overrideSelector(getOverriddenFeatureFlags, {
+      inColab: true,
+    });
+    createComponent();
+    const component = getComponent();
+
+    const rows = component.querySelectorAll('tr');
+    expect(rows.length).toEqual(2);
+  });
+
+  it('creates table data for non editable flags and mat-selects for editable flags', () => {
+    store.overrideSelector(getDefaultFeatureFlags, {
+      inColab: false,
+      enabledExperimentalPlugins: [] as string[],
+    } as FeatureFlags);
+    store.overrideSelector(getOverriddenFeatureFlags, {
+      inColab: true,
+    });
+    createComponent();
+    const component = getComponent();
+
+    const dataCells = component.querySelectorAll('td');
+    expect(dataCells.length).toEqual(3);
+    const selectors = component.querySelectorAll('mat-select');
+    expect(selectors.length).toEqual(1);
```

Comment:
Would it be worth to make the tests a litlte stricter - ie actually verify that the "inColab" row has a mat-select while the "enabledExperimentalPlugins" is marked as unsupported? The test is unusually vague about this.

Reply:
You're right, this is pretty vague. I've gone ahead and added another `expect` checking the value of the `td` which I think should be specific enough.